### ISSUE TITLE
theme(panel_grid_*_x) is interpreted after coord_flip

### ISSUE THREAD

**Issue by TyberiusPrime** [has2k1/plotnine#386]
Similar to #216.
The theme elements are currently evaluated after the coord_flip.
So if you add coord_flip to a plot, you need to change your themes -> bug.

**Comment by has2k1** [comment]
I do not think this is a bug, you can expect either behaviour. `coord_flip` only makes sense because for some geoms (e.g. `geom_bar`) you are restricted to the orientation that you map to, it is not just for convenience. Theming has no restrictions, you can do change the `x` and `y` attributes alike.

Also, although I am not sure if when we will change the stats & geoms to be orientation independent but that will further reduce the need for `coord_flip`.

**Comment by TyberiusPrime** [comment]
I will argure that it is a bug, just like it was for geom_rug:
Coord_flip should be a simple toggle. With the current behaviour you need two change to places. And it is also inconsistent, because you now have axes descriptions in your plot Def that are flipped and those that are not.
This is quite confusing.

**Comment by has2k1** [comment]
Your are right, from an API perspective it is a wrong. However the future is to make geoms and stats orientation independent and that would the use of `coord_flip` irrelevant. Also, `coord_flip` fails to meet expectations in other ways, so if it is to stay in the long term it maybe deprecated or just leave it for sentimental reasons.

**Comment by has2k1** [comment]
Marked as won't fix, because it would require the theme to be aware of coordinate details. This would complicate the code.
